src/bundle_list.c

```diff
@@ -257,7 +257,11 @@ static enum swupd_code show_included_bundles(char *bundle_name, int version)
 
 	if (verbose) {
 		ret = subscription_get_tree(bundles, &subs, mom, true, 0);
+		if (ret == add_sub_NEW) {
```

Comment:
This looks incorrect. What about other values of subscription_get_tree() return?

Reply:
the possible values are these:
- `0` : nothing valid was found, so nothing will be listed. 0 is already the same as `SWUPD_OK`.
- `add_sub_BADNAME` : an invalid bundle was found (a bundle that is not in the MoM), this condition will never happen during `bundle-list --deps` because at this point in the code the bundle was already validated.
- `add_sub_ERR` : the manifest of a bundle could not be downloaded. This is an important error we want to avoid since it will mean the list will be incomplete.
- `add_sub_NEW` : we have gone through at least one valid bundle.

So the only two valid conditions for us at this point is either `0` or `add_sub_NEW`, we should fail if we find `add_sub_ERR`, and `add_sub_BADNAME ` will never happen.

Reply:
show_included_bundles() is suppose to return a ```enum swupd_code```. In the case you described you are possibly returning the ```add_sub_``` values you don't want.
This can lead to swupd returning the error codes with meanings that aren't listed on the manual.

One suggestion to avoid this kind of problems is never using the same variable for return codes with different meanings, even if they have the same type. The ideal here would be to have one exit_code with type ```enum swupd_code``` and another with type int for the int return codes.

Reply:
ahh you are right, we don't want to propagate those error codes up, I overlooked that, I'll correct that. Thanks for pointing that out.